Tests/Linq/UserTests/Issue3161Tests.cs

```diff
@@ -112,10 +112,9 @@ public void CrossApplyTwice([IncludeDataSources(TestProvName.AllSqlServer2008Plu
 				.ToList();
 			Assert.That(ret.Count, Is.EqualTo(2));
 
-			// uncomment after 3161 fixed
-			//var baselines = GetCurrentBaselines();
-			//baselines.Should().Contain("SELECT", Exactly.Twice());
-			//baselines.Should().Contain("SELECT TOP", Exactly.Once());
+			var baselines = GetCurrentBaselines();
+			baselines.Should().Contain("SELECT", Exactly.Times(3));
+			baselines.Should().Contain("SELECT TOP", Exactly.Twice());
```

Comment:
Without the fix, "SELECT" occurs 3 times and "SELECT TOP" occurs once.  It would seem it it could be coincidental that these checks serve to validate that the query is executed as a single statement.  I suggest adding additional tests here along with comments to reinforce the issue we are trying to test for.

```suggestion
			// validate that the prior statement executed as a single query, not two distinct queries
			var baselines = GetCurrentBaselines();
			baselines.Should().Contain("SELECT", Exactly.Times(3));
			baselines.Should().Contain("SELECT TOP", Exactly.Twice());

			// LastQuery will only return a single query, so if it was split into two queries, not all name fields would be present
			var lastQuery = ((DataConnection)db).LastQuery;
			lastQuery.Should().Contain("NAME1", Exactly.Once());
			lastQuery.Should().Contain("NAME2", Exactly.Once());
			lastQuery.Should().Contain("NAME3", Exactly.Once());
```

Reply:
Note that `using LinqToDB.Data;` is required to be added for the above change.